recipe_scrapers/sunset.py

```diff
@@ -1,7 +1,40 @@
 from ._abstract import AbstractScraper
+from ._grouping_utils import IngredientGroup
 
 
 class Sunset(AbstractScraper):
     @classmethod
     def host(cls):
         return "sunset.com"
+
+    def ingredient_groups(self):
+        groups = []
+        current_group = []
+        current_purpose = None
+        seen_ingredients = set()
+
+        for element in self.soup.select(".cooked-single-ingredient"):
+            classes = element.get("class", [])
+            if "cooked-heading" in classes:
+                if current_group:
+                    groups.append(
+                        IngredientGroup(
+                            ingredients=current_group, purpose=current_purpose
+                        )
+                    )
+                current_group = []
+                current_purpose = element.get_text(strip=True)
+            elif "cooked-ingredient" in classes:
+                ingredient = " ".join(
+                    element.get_text(separator=" ", strip=True).split()
+                )
+                if ingredient not in seen_ingredients:
```

Comment:
The page returns 2 of each ingredient which requires the dupe check. I've tried using a more specific selector and still get duplicates for both test cases. 

This could be a possible candidate for use outside of equipment for the `get_equipment` as discussed in #1321

Reply:
Hmm, OK.  Instead of adding more specific filtering on the ingredients already found by the selector, could we adjust the top-level selector to narrow to only one of the lists of ingredients on the page?

To maybe explain that another way, I _think_ this should work - it does for the test cases when I run it here:

```diff
-        for element in self.soup.select(".cooked-single-ingredient"):
+        for element in self.soup.select(".recipe-ingredients .cooked-single-ingredient"):
```

Reply:
That fixed it, not sure why I wasn't able to get it work before now but I'll push this change now!